draft-ZIP-0227.rst

```diff
@@ -0,0 +1,184 @@
+
+- The issuer knows in advance the receivers of the issued asset.
+- The asset is of non-fungible type, where each asset type can be made part of a single “series”
+- The supply of the asset is limited or not
+- The assets can be wrapped versions of assets in other chains (as long as there is a bridge that supports it)
+
+The protocol is as follows:
+
+- For each asset type issued, generate a sequence of output notes, each with the following fields:
+    - ZSA output note :math:`\mathsf{ note =(d, pkd, v, \rho, \psi, \mathsf{type}_\mathbb{P}, rcm)}`, where :math:`\mathsf{type}_\mathbb{P} := \mathsf{GroupHash^{\mathbb{P}}}\mathsf{(assetID)}`
+- Generate commitment of the note. `cm` as
+
+:math:`cm = \mathsf{NoteCommit^{ZSA}_{rcm}}(\mathsf{repr\mathbb{_P}(g_d)}, \mathsf{repr\mathbb{_P}(pk_d)}, v, \rho, \psi,\mathsf{type}_\mathbb{P})`
+
+- Sign the issuance note with the `issuerID` as the signing key, using RedPallas as its signature scheme, on the `SIGHASH` of the transaction. Note that the `SIGHASH` will change as we include a new bundle in the Zcash transaction to enable this issuance mechanism on chain.
+
+Consensus Changes
+-----------------
+
+Issuance requires the following additions to the global state: 
+
+First, Zcash clients must keep a mapping `issuanceSupplyInfos` from `assetID` to `issuanceSupplyInfo := (totalSupply, finiteSupply)`.
+
+**Consensus rules**
+
+For each issuance instance,
+
+- Check that `assetID` is properly constructed,
+- Check that each commitment `cm` is properly constructed from each “output” note,
+- Check that the issuance instance has a valid signature based on the `issuerID`,
+- Check that :math:`\mathsf{type}_\mathbb{P}` is properly projected into the correct curve from `assetID`. That is, that :math:`\mathsf{type}_\mathbb{P} :=  \mathsf{GroupHash^{\mathbb{P}}}\mathsf{(assetID)}`
+- If `issuanceSupplyInfo[assetID].finiteSupply`is true, reject. Otherwise, set `issuanceSupplyInfo[assetID].finiteSupply = finiteSupply` .
+- Add the sum of all the values in all output notes per `assetID` to `issuanceSupplyInfo[assetID].totalSupply`
+- After verification, add note commitment to the merkle tree.
+
+Other Considerations
+====================
+
+Fee Structures
+--------------
+
+The fee mechanism described in this ZIP may be replaced by a future “Fee Mechanism” ZIP proposal. However, we do need to have some basic fees for the Issuance protocol, as it differs from the usual transfer mechanism in that it adds a whole new structure to the transaction bundle, it adds a new global structure for validators to keep in memory and in turn it adds new consensus rules that are independent of previous structures.
```

Comment:
```suggestion
The fee mechanism described in this ZIP may be replaced by a future “Fee Mechanism” ZIP proposal. However, we do need to have some basic fees for the Issuance protocol, as it differs from the usual transfer mechanism in that it adds a whole new structure to the transaction bundle, it adds a new global structure for validators to keep in memory, and in turn, it adds new consensus rules that are independent of previous structures.
```

Reply:
Removed this as a the Fees zip is written.